### ISSUE TITLE
Helm charts

### ISSUE THREAD

**Issue by juliusvonkohout** [kubeflow/manifests#2730]
### Validation Checklist

- [X] Is this a Kubeflow issue?
- [X] Are you posting in the right repository ?
- [X] Did you follow the installation guide https://github.com/kubeflow/manifests?tab=readme-ov-file ?
- [X] Is the issue report properly structured and detailed with version numbers?
- [X] Is this for Kubeflow development ?
- [ ] Would you like to work on this issue?
- [X] Join our slack channel using [wg-manifests](https://www.kubeflow.org/docs/about/community/).

### Version

master

### Describe your issue

Migration of https://github.com/kubeflow/kubeflow/issues/3173

@kromanow94 @AndersBennedsgaard @alex-treebeard @clayrisser @Syntax3rror404 @bhack @naseemkullah

### Steps to reproduce the issue

Search for a helm chart

**Comment by kromanow94** [comment]
I'm continuing the discussion that sparked here:
https://github.com/kubeflow/manifests/issues/2451#issuecomment-2213603123

Hey @doncorsean , that's so great! I love what you're doing!

Lets use this occasion so I can update you with what is happening right now in the Helm Chart:
* We're finalizing work around having the kserve and knative installed (standard approach, kserve and knative should be installed as external dependencies and some integration manifests are put in the KF Helm Chart)
  * knative
    * quickstart scripts are configured to install knative operator either from the https://github.com/knative/operator/releases/download/knative-v1.11.0/operator.yaml directly or from the kustomize (with ArgoCD) in https://github.com/knative/operator/tree/knative-v1.11.0/config/default (depending if we're using ArgoCD or plain Helm approach)
      * I didn't find any helm chart for the knative so we probably have to live with it how it is for now
    * integration manifests are configured to deploy `KnativeEventing` and `KnativeServing` CRDs
  * kserve
    * quickstart scripts are installing kserve either from:
      * locally downloaded kserve helm chart (because the 0.11.2 kserve is not available in the Helm Chart repo, only 0.12.0+: https://github.com/kserve/kserve/pkgs/container/charts%2Fkserve)
      * ArgoCD App pointing to the kserve git repo where helm charts are defined (https://github.com/kserve/kserve/tree/v0.11.2/charts)
* What is still missing and we're be happy to welcome a contribution:
  * KServe Models Web App (for this let's wait until the kserve/knative thing is officially released in the github/kromanow94/kubeflow-manifests)
  * Katib
  * Tensorboard Controller/Tensorboards Web App
  * Training Operator

If you're interested, we can talk at the next Kubeflow Platform Meeting (Manifests + Security WG) and have some alignment on what would be best to pick up for your contribution :).

**Comment by doncorsean** [comment]
Sounds good @kromanow94, I'll take a look in Slack and get dialed into the meeting. Is it on the calendar already @juliusvonkohout?

**Comment by juliusvonkohout** [comment]
#kubeflow-platform is the channel

**Comment by Syntax3rror404** [comment]
ping

**Comment by kromanow94** [comment]
Just for update, we're still working on the Helm Chart :). For `1.8.1` we need just a couple of things. This is Tensorboards and a separate Helm Chart for CRDs because they became huge. I think we're going to have a ready version sometime in November :).

**Comment by kromanow94** [comment]
Hey, a new release for Kubeflow Helm Chart 0.3.0 just dropped!

https://github.com/kromanow94/kubeflow-manifests/releases/tag/kubeflow-0.3.0

See the release notes for more information and quickstart example.

Key Changes:
* TensorBoard Integration.
* Separate Helm Chart for All CRDs.
* Temporary usage Helm Repository Mirror for Knative Operator until `knative-operator` is served on their official repository.
* Simplified Knative Integration.

Also, not too long ago we did a small release `0.2.1` where we added Katib integration with help from @doncorsean! Thank you for your time and effort!

Next goals:
* Add Models Web App to the Helm Chart.
* Update Istio integration to include latest changes included in Kubeflow 1.9.1 for simplified and improved M2M Token and in-cluster Self Signed OIDC Issuer.

After we have the two next goals reached, we're going to ask people to test and provide feedback. After that we're hoping to make an official release for the Kubeflow Helm Chart!

**Comment by chasecadet** [comment]
This is incredible! End users have been asking for this!

**Comment by kromanow94** [comment]
Hey, a new release for Kubeflow Helm Chart 0.4.0 just dropped!

This release finalizes the technical foundation for a full-fledged Kubeflow installation using only Helm Charts for its components and dependencies. 🎉

We still lack some documentation but from technical perspective, this is all what's needed to have a Kubeflow 1.8.1 deployed with just the Helm Charts!

For more information, see this release: [Kubeflow Helm Chart v0.4.0](https://github.com/kromanow94/kubeflow-manifests/releases/tag/kubeflow-0.4.0).

Also, feel free to join the discussion on the CNCF Slack Channel `#kubeflow-helm-chart`:
https://cloud-native.slack.com/archives/C083H8QCAE8

**Comment by juliusvonkohout** [comment]
@kromanow94 it is going on in https://github.com/kubeflow/community/pull/830 and with GOC

**Comment by andreyvelich** [comment]
@juliusvonkohout Please can we transfer this issue to the `kubeflow/community` repo since KEP number must be equal to issue created in the same repo.
For example: KEP-819: https://github.com/kubeflow/community/issues/819

**Comment by juliusvonkohout** [comment]
We can do so, but for the implementation here i would have to open up a new issue then.

**Comment by andreyvelich** [comment]
Yes, we can open 2 separate issues. However, we also need to agree where Helm Charts will be developed.
We might need to create a new repo for it, similar to other projects:
https://github.com/grafana/helm-charts
https://github.com/prometheus-community/helm-charts
https://github.com/argoproj/argo-helm

**Comment by juliusvonkohout** [comment]
> Yes, we can open 2 separate issues. However, we also need to agree where Helm Charts will be developed. We might want to consider to create a new repo for it, similar to other projects: https://github.com/grafana/helm-charts https://github.com/prometheus-community/helm-charts https://github.com/argoproj/argo-helm

There are many hard technical reasons against a new repository. We need to have them next to each other to share all the gha workflows and tests and keep them synchronized with a single source of truth.
